### PR TITLE
Remove dead PHPStan rules

### PR DESCRIPTION
**Pull request by carusogabriel**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -3,57 +3,46 @@ parameters:
...
 
         # false positive
-        - '#Parameter \#1 \$methodCallNode of method Rector\\NodeAnalyzer\\MethodArgumentAnalyzer::isMethodSecondArgumentNull\(\) expects PhpParser\\Node\\Expr\\MethodCall, PhpParser\\Node given#'
-        - '#Parameter \#1 \$kernelClass of method Rector\\Bridge\\Symfony\\SymfonyKernelParameterGuard::(ensureKernelClassExists|ensureIsKernelInstance)\(\) expects string, string\|null given#'
+        - '#Parameter \#1 \$kernelClass of method Rector\\Bridge\\Symfony\\SymfonyKernelParameterGuard::(ensureKernelClassExists|ensureIsKernelInstance)\(\) expects string, string\|null given#' # 2
 
         # intentionally incorrect - part of the test
-        - '#Parameter \#2 \$codeSamples of class Rector\\RectorDefinition\\RectorDefinition constructor expects array<Rector\\RectorDefinition\\CodeSample>, array<int, stdClass> given#'
+        - '#Parameter \#2 \$codeSamples of class Rector\\RectorDefinition\\RectorDefinition constructor expects array<Rector\\RectorDefinition\\CodeSample>, array<int, stdClass> given#' # 1
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply phpstan.neon]: That's rather automatic job for a script

File: src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php [hunk 2]
```diff
@@ -91,8 +91,6 @@ public function default()
 
     public function isCandidate(Node $node): bool
     {
-        $this->activeVariableInfo = null;
```
Comment (TomasVotruba) [inline src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: Why?

File: phpstan.neon (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply phpstan.neon]: Or even better PHPStan feature request

File: src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php [hunk 3]
```diff
@@ -22,7 +22,7 @@
     private $variablesToPropertyFetchCollection;
 
     /**
-     * @var VariableInfo|null
+     * @var VariableInfo
```
Comment (TomasVotruba) [inline src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: This is not valid

Comment (carusogabriel) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: Why not?

File: src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php (hunk 2 above)
Comment (carusogabriel) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: Isn’t necessary, we’ve already initialized the property when declared it

File: src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php (hunk 3 above)
Comment (TomasVotruba) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: It's often null

File: src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php (hunk 2 above)
Comment (TomasVotruba) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: It changed on each iteration, it's not one-time action.

Often it's not null.

Comment (carusogabriel) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: So we should change the implementation. Can you handle it?

Comment (TomasVotruba) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: Why would you change it? It works great now

Comment (carusogabriel) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: So let's keep it. Maybe we should remove PHPStan in the future than :)

Comment (TomasVotruba) [reply src/Rector/Architecture/DependencyInjection/ReplaceVariableByPropertyFetchRector.php]: Why?

**Comment by TomasVotruba** [comment]
:+1: Thank you

**Pull request merged**